core/lib/lantern-trace-saver.js

```diff
@@ -140,17 +170,43 @@ function convertNodeTimingsToTrace(nodeTimings) {
       priority: record.priority,
     };
 
+    const {dnsResolutionTime, connectionTime, sslTime, timeToFirstByte} = timing.connectionTiming;
+    let sslStart = -1;
+    let sslEnd = -1;
+    if (connectionTime !== undefined && sslTime !== undefined) {
+      sslStart = connectionTime - sslTime;
+      sslEnd = connectionTime;
+    }
     const receiveResponseData = {
       ...requestData,
       statusCode: record.statusCode,
       mimeType: record.mimeType,
       encodedDataLength: record.transferSize,
       fromCache: record.fromDiskCache,
       fromServiceWorker: record.fetchedViaServiceWorker,
+      timing: {
+        requestTime: toMicroseconds(startTime) / (1000 * 1000),
+        dnsStart: dnsResolutionTime === undefined ? -1 : 0,
+        dnsEnd: dnsResolutionTime ?? -1,
+        connectStart: dnsResolutionTime ?? -1,
+        connectEnd: connectionTime ?? -1,
+        sslStart,
+        sslEnd,
+        sendStart: connectionTime ?? 0,
+        sendEnd: connectionTime ?? 0,
+        receiveHeadersEnd: timeToFirstByte,
+        workerStart: -1,
+        workerReady: -1,
+        proxyStart: -1,
+        proxyEnd: -1,
+        pushStart: 0,
+        pushEnd: 0,
+      },
```

Comment:
@paulirish you might have lots of thoughts. Only so much we can do with the timings we have, so it's a bit handwavey (especially `sendStart`/`sendEnd`/`receiveHeadersEnd`), but maybe you have ideas

Reply:
yeah this looks great